File: Chagok/frontend/src/types/calendar.ts
/**
 * Calendar Types
 * 003-role-based-ui Feature - US7
 *
 * Type definitions for calendar events and related data.
 */

export type CalendarEventType = 'court' | 'meeting' | 'deadline' | 'internal' | 'other';

export interface CalendarEvent {
  id: string;
  user_id: string;
  case_id?: string;
  case_title?: string;
  title: string;
  description?: string;
  event_type: CalendarEventType;
  start_time: string;
  end_time?: string;
  location?: string;
  reminder_minutes?: number;
  color?: string;
  created_at?: string;
  updated_at?: string;
}

export interface CalendarEventCreate {
  title: string;
  event_type: CalendarEventType;
  start_time: string;
  end_time?: string;
  description?: string;
  location?: string;
  case_id?: string;
  reminder_minutes?: number;
}

export interface CalendarEventUpdate {
  title?: string;
  event_type?: CalendarEventType;
  start_time?: string;
  end_time?: string;
  description?: string;
  location?: string;
  case_id?: string;
  reminder_minutes?: number;
}

export interface CalendarEventsResponse {
  events: CalendarEvent[];
  total: number;
}

export interface UpcomingEventsResponse {
  events: CalendarEvent[];
}

export interface RemindersResponse {
  reminders: CalendarEvent[];
}

export interface EventTypeColors {
  court: string;
  meeting: string;
  deadline: string;
  internal: string;
  other: string;
}

// Event type color mapping (matches backend)
export const EVENT_TYPE_COLORS: EventTypeColors = {
  court: '#ef4444',     // Red
  meeting: '#3b82f6',   // Blue
  deadline: '#f59e0b',  // Amber
  internal: '#8b5cf6',  // Purple
  other: '#6b7280',     // Gray
};

// Korean labels for event types
export const EVENT_TYPE_LABELS: Record<CalendarEventType, string> = {
  court: '재판',
  meeting: '상담',
  deadline: '마감',
  internal: '내부회의',
  other: '기타',
};

File: Chagok/backend/app/db/schemas/calendar.py
"""
Calendar Event Schemas
"""

from pydantic import BaseModel, Field
from typing import Optional, List
from datetime import datetime
from app.db.models import CalendarEventType


# ============================================
# Calendar Event Schemas
# ============================================
class CalendarEventCreate(BaseModel):
    """Calendar event creation request schema"""
    case_id: Optional[str] = None
    title: str = Field(..., min_length=1, max_length=255)
    description: Optional[str] = None
    event_type: CalendarEventType = CalendarEventType.OTHER
    start_time: datetime
    end_time: Optional[datetime] = None
    location: Optional[str] = None
    reminder_minutes: int = 30


class CalendarEventUpdate(BaseModel):
    """Calendar event update request schema"""
    case_id: Optional[str] = None
    title: Optional[str] = Field(None, min_length=1, max_length=255)
    description: Optional[str] = None
    event_type: Optional[CalendarEventType] = None
    start_time: Optional[datetime] = None
    end_time: Optional[datetime] = None
    location: Optional[str] = None
    reminder_minutes: Optional[int] = None


class CalendarEventOut(BaseModel):
    """Calendar event output schema"""
    id: str
    user_id: str
    case_id: Optional[str] = None
    case_title: Optional[str] = None  # Joined from Case table
    title: str
    description: Optional[str] = None
    event_type: CalendarEventType
    start_time: datetime
    end_time: Optional[datetime] = None
    location: Optional[str] = None
    reminder_minutes: int
    created_at: datetime

    class Config:
        from_attributes = True


class CalendarEventListResponse(BaseModel):
    """Calendar event list response schema"""
    events: List[CalendarEventOut]
    total: int
